backend/prompt_studio/prompt_studio_index_manager_v2/models.py

```diff
@@ -0,0 +1,130 @@
+from utils.user_context import UserContext
+
+logger = logging.getLogger(__name__)
+
+
+class IndexManager(BaseModel):
+    """Model to store the index details."""
+
+    index_manager_id = models.UUIDField(
+        primary_key=True, default=uuid.uuid4, editable=False
+    )
+
+    document_manager = models.ForeignKey(
+        DocumentManager,
+        on_delete=models.CASCADE,
+        related_name="index_managers",
+        editable=False,
+        null=False,
+        blank=False,
+    )
+
+    profile_manager = models.ForeignKey(
+        ProfileManager,
+        on_delete=models.SET_NULL,
+        related_name="index_managers",
+        editable=False,
+        null=True,
+        blank=True,
+    )
+
+    raw_index_id = models.CharField(
+        db_comment="Field to store the raw index id",
+        editable=False,
+        null=True,
+        blank=True,
+    )
+
+    summarize_index_id = models.CharField(
+        db_comment="Field to store the summarize index id",
+        editable=False,
+        null=True,
+        blank=True,
+    )
+
+    index_ids_history = models.JSONField(
+        db_comment="List of index ids",
+        default=list,
+        null=False,
+        blank=False,
+    )
+
+    created_by = models.ForeignKey(
+        User,
+        on_delete=models.SET_NULL,
+        related_name="index_managers_created",
+        null=True,
+        blank=True,
+        editable=False,
+    )
+
+    modified_by = models.ForeignKey(
+        User,
+        on_delete=models.SET_NULL,
+        related_name="index_managers_modified",
+        null=True,
+        blank=True,
+        editable=False,
+    )
+
+    class Meta:
+        verbose_name = "Index Manager"
+        verbose_name_plural = "Index Managers"
+        db_table = "index_manager_v2"
+        constraints = [
+            models.UniqueConstraint(
+                fields=["document_manager", "profile_manager"],
+                name="unique_document_manager_profile_manager_index",
+            ),
+        ]
+
+
+def delete_from_vector_db(index_ids_history, vector_db_instance_id):
+    organization_identifier = UserContext.get_organization_identifier()
+    util = PromptIdeBaseTool(log_level=LogLevel.INFO, org_id=organization_identifier)
+    vector_db = VectorDB(
+        tool=util,
+        adapter_instance_id=vector_db_instance_id,
+    )
+    for index_id in index_ids_history:
+        logger.debug(f"Deleting from VectorDB - index id: {index_id}")
+        try:
+            vector_db.delete(ref_doc_id=index_id)
+        except Exception as e:
+            # Log error and continue with the next index id
+            logger.error(f"Error deleting index: {index_id} - {e}")
+
+
+# Function will be executed every time an instance of IndexManager is deleted.
+@receiver(pre_delete, sender=IndexManager)
+def perform_vector_db_cleanup(sender, instance, **kwargs):
+    """Signal to perform vector db cleanup."""
+    logger.info("Performing vector db cleanup")
```

Comment:
@muhammad-ali-e Please enrich this log message with the vector db id too, if we need this.

Reply:
updated